### PR TITLE
[AKS] Add mutable fips cli flags (enable/disable on nodepool update)

### PR DESCRIPTION
**Pull request by pineapplethevoyager**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by FumingZhang — COMMENTED** [review]
Looks good to me, but seems you accidentally touched the recording file `test_aks_skip_gpu_driver_install.yaml` and failed the CI, please reset the change on that file

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=97029417&view=results) to validate the change, test passed!
- test_aks_create_update_fips_flow

**Review by FumingZhang — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -4645,6 +4645,109 @@ def test_aks_create_update_vtpm_flow(self, resource_group, resource_group_locati
...
+            "--enable-fips-image"
+        )
+        self.cmd(
+            create_cmd,
+            checks=[
+                self.check("provisioningState", "Succeeded"),
+                self.check("agentPoolProfiles[0].enableFips", True),
+            ],
+        )
+
```
Comment (chengliangli0918) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:4686]: add an update without either --enable-fips-image or --disable-fips-image, EnableFips should not change.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -4645,6 +4645,109 @@ def test_aks_create_update_vtpm_flow(self, resource_group, resource_group_locati
...
+            "--cluster-name={name} "
+            "--name={node_pool_name_second} "
+            "--os-type Linux "
+            '--aks-custom-headers AKSHTTPCustomFeatures=Microsoft.ContainerService/MutableFipsPreview ',
+            checks=[
+                self.check("provisioningState", "Succeeded"),
+                self.check("enableFips", False),
+            ],
+        )
+
```
Comment (chengliangli0918) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:4722]: add an update without either --enable-fips-image or --disable-fips-image, EnableFips should not change.

Comment (pineapplethevoyager) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:4722]: done

**Pull request merged**